Fix test pid_ns_reap

bernhardu opened the pull request:
Is there a reason not to continue here?

Shows up in CI:
https://buildkite.com/julialang/rr/builds/1220#01882448-dc7e-4e6f-89fd-ddefa9dbf22b/4360-7958

bin/rr record bin/pid_ns_reap

Avoids error message:
```
  [FATAL src/AutoRemoteSyscalls.cc:659:retrieve_fd_arch() errno: EBADF]
   (task 693273 (rec:693273) at time 308)
   -> Assertion `errno == ENOSYS' failed to hold. Failed in pidfd_getfd errno=EBADF
```
Affected tests:
  pid_ns_reap

rocallahan commented:
We need to understand what the underlying bug is, we shouldn't just patch over it.

bernhardu committed:
Avoid retrieval of desched_fd from a zombie process.

Shows up in CI:
  https://buildkite.com/julialang/rr/builds/1266#01895379-a3f0-4778-9bfa-0874291c4f9a/1834-7500

Reproducible by:
  rr record pid_ns_reap

Error message:
 [FATAL src/AutoRemoteSyscalls.cc:658:retrieve_fd_arch() errno: EBADF]
  (task 659770 (rec:659770) at time 308)
  -> Assertion `errno == ENOSYS' failed to hold. Failed in pidfd_getfd errno=EBADF

bernhardu commented:
> We need to understand what the underlying bug is, we shouldn't just patch over it.

I tried to have another look at the process for which we are able to do the `pidfd_open` but the `pidfd_getfd` fails with EBADF.

And I noticed this process is listed as zombie.

Is the `pidfd_getfd` supposed to work on a zombie process (or even needed at that time)?

I pushed a new patch to avoid `pidfd_getfd` for zombie processes.

bernhardu commented:
Got fixed by 0e37796a451e, thanks.
